### DISCUSSION TITLE
Marks support for list item markers (bullets, numbers, letters)

### DISCUSSION THREAD

**Discussion by arekdygas** [ueberdosis/tiptap#4117] [General Questions] [answered]
Hi,

When setting marks on bulleted/ordered list items, these marks only apply to list item text, and don't affect the item marker. For example, setting bold on ordered list items, only makes texts bold, while numbers stay non-bold.

I'd like it to apply the mark the number or bullet itself, i.e. to the whole `LI`, or maybe just to `::marker` pseudo-element, so the whole marked text remain consistent, e.g. the same color, font-family or size.

As far as I understand the code, when I select list item in editor, and execute `setMark` command, the mark is added to the parent of inline elements. In case of ListItem, this is paragraph, which is a parent of text node. What would be the best option, to extend this mark to ListItem itself?

**Comment by KevinTss** [comment]
I think is the same issue I have with custom node, seems like marks are not supported in custom nodes, maybe a link whit [this](https://github.com/yjs/y-prosemirror/pull/135)?

**Reply by KevinTss** [reply]
Any idea @philipisik, @svenadlung  ?

**Reply by philipisik** [reply]
Or perhaps @bdbch.

**Reply by bdbch** [reply]
Could you explain this a bit more? Did you mean custom node views or custom html elements (webcomponents)?

**Comment by svenadlung** [comment, marked as answer]
Tbh  I don't think you can solve this through marks. The ListItem is a node and the marker is controlled by it. Without thinking very much about it, I would extend the ListItem and add attributes, which should be updated to affect the styling of the `<li>`s accordingly.

**Reply by bdbch** [reply]
Pretty much this. As you said @arekdygas the marks will be added around the inline text (which is the text inside your paragraph node). This won't update the paragraph itself nor the list item.

I guess you want to make a whole list item bold / italic and want the list-style-marker to reflect that? I guess it won't work that easy with just `setMark`.

You could extend the listItem and add attributes and a command to set marks inside a list item. Inside that command, you could check if just a part of the list text is selected - if so just call `setMark` – otherwise you could set the attribute of your list item to reflect the style.

But that will also cause issues if you then try to toggle the mark inside the inline text which will do nothing except you also add custom logic for that.

**Reply by arekdygas** [reply]
Yes - I thought a bit about this since I asked the question, and came to the same conclusion. Thanks for the confirmation! :)
